app/scripts/metamask-controller.js

```diff
@@ -4232,6 +4303,36 @@ export default class MetamaskController extends EventEmitter {
     // append selectedNetworkClientId to each request
     engine.push(createSelectedNetworkMiddleware(this.controllerMessenger));
 
+    const { selectedNetworkClientId } = this.networkController.state;
+
+    const selectedNetworkClientIdForDomain =
+      this.selectedNetworkController.getNetworkClientIdForDomain(origin);
+
+    // Not sure that this will happen anymore
```

Comment:
will it 😅 ?

Reply:
It is currently possible, since getNetworkClientIdForDomain can return undefined. I put this comment here cause im not sure if that ever happens in practice, but the typings suggest that it can be possible.

Reply:
Ill add this to the list of things to follow up on once we get this out.